src/count-islands/index.js

```diff
@@ -1,3 +1,37 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  let islands = 0;
+
+  let width = grid.length;
+
+  let height = grid[0].length;
+
+  for (let i = 0; i < width; i++) {
+    for (let j = 0; j < height; j++) {
+      if (grid[i][j] == "1") {
+        depthFirstSearch(grid, i, j);
+        islands++;
+      }
+    }
+  }
+  return islands;
+
+  function depthFirstSearch(grid, i, j) {
```

Comment:
move this depthFirstSearch() function out of the countIsLand() function a function should have a single responsibility